controllers/snapshotenvironmentbinding_controller.go

```diff
@@ -157,12 +157,13 @@ func addTargetIfNotExists(secret *rapi.RemoteSecret, target rapi.RemoteSecretTar
 }
 
 func removeTarget(secret *rapi.RemoteSecret, target rapi.RemoteSecretTarget) {
-	for idx := range secret.Spec.Targets {
-		existingTarget := secret.Spec.Targets[idx]
-		if targetsMatch(existingTarget, target) {
-			secret.Spec.Targets = append(secret.Spec.Targets[:idx], secret.Spec.Targets[idx+1:]...)
+	tmp := secret.Spec.Targets[:0]
```

Comment:
This is creating an empty slice backed by the `secret.Spec.Target` array. You then append to this slice which IMHO just creates a new array under the hood. So how about just creating a new array here with the same capacity as the targets?

E.g.:
```go
tmp := make([]rapi.RemoteSecretTarget, 0, len(secret.Spec.Targets))
```

This will avoid re-allocations of the `tmp` array (and will waste at most 1 element in it).

Reply:
ok